Source/CarthageKit/FrameworkExtensions.swift

```diff
@@ -261,6 +261,13 @@ extension NSURL {
 
 		return .Failure(.ReadFailed(self, error))
 	}
+
+	public func hasSubdirectory(possibleSubdirectory: NSURL) -> Bool {
+		if scheme == possibleSubdirectory.scheme, let path = self.pathComponents, otherPath = possibleSubdirectory.pathComponents where path.count <= otherPath.count {
+			return Array(otherPath[0..<path.count]) == path
```

Comment:
I [mentioned here](https://github.com/Carthage/Carthage/pull/892#discussion-diff-47987373) that `hasPrefix` isn't safe because it's hard to ensure that you're not matching `Carthage-blah` with `hasPrefix("Carthage")`.